middleman_protocol/tests/test_helpers.py

```diff
@@ -0,0 +1,353 @@
+                    assertpy.assert_that(deserialized_message.payload).is_instance_of(Ping)
+                    assertpy.assert_that(deserialized_message.payload).is_equal_to(payload)
+
+    def test_that_receiving_encoded_message_should_decode_on_the_fly(self):
+        middleman_message = GolemMessageFrame(Ping(), self.request_id)
+        raw_message = append_frame_separator(
+            middleman_message.serialize(private_key=CONCENT_PRIVATE_KEY)
+        )
+
+        raw_message = raw_message[:10] + ESCAPE_CHARACTER + raw_message[len(ESCAPE_CHARACTER) + 10:]
+        raw_message_encoded = escape_encode_raw_message(raw_message)
+
+        assert FRAME_SEPARATOR not in raw_message_encoded
+
+        raw_message_encoded = append_frame_separator(raw_message_encoded)
+
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as server_socket:
+            server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+            with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as client_socket:
+                client_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+                server_socket.bind(('127.0.0.1', 8001))
+                server_socket.listen(1)
+
+                client_socket.connect(('127.0.0.1', 8001))
+
+                (connection, _address) = server_socket.accept()
+
+                client_socket.send(raw_message_encoded)
+                raw_message_received = next(unescape_stream(connection=connection))
+
+        assertpy_bytes_starts_with(raw_message, raw_message_received)
+        assertpy.assert_that(len(raw_message_received)).is_greater_than_or_equal_to(FRAME_PAYLOAD_STARTING_BYTE)
+
+    def test_that_receiving_wrongly_encoded_message_should_return_none(self):
+        middleman_message = GolemMessageFrame(Ping(), self.request_id)
+        raw_message = middleman_message.serialize(
+            private_key=CONCENT_PRIVATE_KEY,
+        )
+
+        raw_message_encoded = escape_encode_raw_message(raw_message)
+        raw_message_encoded = raw_message_encoded + ESCAPE_CHARACTER + b'\xff'
+        raw_message_encoded = append_frame_separator(raw_message_encoded)
+
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as server_socket:
+            server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+            with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as client_socket:
+                client_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+                server_socket.bind(('127.0.0.1', 8001))
+                server_socket.listen(1)
+
+                client_socket.connect(('127.0.0.1', 8001))
+
+                (connection, _address) = server_socket.accept()
+
+                client_socket.send(raw_message_encoded)
+                raw_message_received = next(unescape_stream(connection=connection))
+
+        assertpy.assert_that(raw_message_received).is_equal_to(None)
```

Comment:
for `None` comparison, please use: `assertpy.assert_that(raw_message_received).is_none()`